iroh-embed/src/p2p.rs

```diff
@@ -0,0 +1,128 @@
+//! The p2p service to use in an iroh system.
+
+use std::path::PathBuf;
+
+use anyhow::Result;
+use iroh_one::mem_p2p;
+use iroh_p2p::{Config as P2pConfig, Libp2pConfig};
+use iroh_rpc_types::p2p::P2pAddr;
+use iroh_rpc_types::store::StoreAddr;
+use iroh_rpc_types::Addr;
+use tokio::task::JoinHandle;
+
+// TODO:
+//
+// - Need to allow configuring in memory keystore
+// - make Lib2p2Config non_exhaustive and provide a builder
+
+/// The iroh peer-to-peer (p2p) service.
+///
+/// An iroh system needs a p2p service to participate in the IPFS network.
+#[derive(Debug)]
+pub struct P2pService {
+    task: JoinHandle<()>,
+    addr: P2pAddr,
+}
+
+impl P2pService {
+    /// Starts a new iroh peer-to-peer service.
+    ///
+    /// This implicitly starts a task on the tokio runtime to manage the storage node.
+    ///
+    /// The `key_store_path` is the directory where the cryptographic identity of this p2p
+    /// node is stored.  This uses the ssh key files format.
```

Comment:
maybe add a note that an identity is generated the first time, and reused if one is found?